packages/backend-secret/src/ssm_secret_with_amplify_error_handling.ts

```diff
@@ -0,0 +1,110 @@
+import { AppId, BackendIdentifier } from '@aws-amplify/plugin-types';
+import {
+  Secret,
+  SecretClient,
+  SecretIdentifier,
+  SecretListItem,
+} from './secret.js';
+import { SecretError } from './secret_error.js';
+import { AmplifyFault, AmplifyUserError } from '@aws-amplify/platform-core';
+import { SSMServiceException } from '@aws-sdk/client-ssm';
+
+/**
+ * Handles errors and translate them to AmplifyUserError or AmplifyFaults
+ * To be used exclusively by the amplify cli
+ */
+export class SSMSecretClientWithAmplifyErrorHandling implements SecretClient {
+  /**
+   * wraps the secretClient with Amplify CLI specific error handling
+   */
+  constructor(private readonly secretClient: SecretClient) {}
+
+  getSecret = async (
+    backendIdentifier: BackendIdentifier | AppId,
+    secretIdentifier: SecretIdentifier
+  ): Promise<Secret> => {
+    try {
+      return await this.secretClient.getSecret(
+        backendIdentifier,
+        secretIdentifier
+      );
+    } catch (e) {
+      throw this.translateToAmplifyError(e, 'Get');
+    }
+  };
+
+  listSecrets = async (
+    backendIdentifier: BackendIdentifier | AppId
+  ): Promise<SecretListItem[]> => {
+    try {
+      return await this.secretClient.listSecrets(backendIdentifier);
+    } catch (e) {
+      throw this.translateToAmplifyError(e, 'List');
+    }
+  };
+
+  setSecret = async (
+    backendIdentifier: BackendIdentifier | AppId,
+    secretName: string,
+    secretValue: string
+  ): Promise<SecretIdentifier> => {
+    try {
+      return await this.secretClient.setSecret(
+        backendIdentifier,
+        secretName,
+        secretValue
+      );
+    } catch (e) {
+      throw this.translateToAmplifyError(e, 'Set');
+    }
+  };
+
+  removeSecret = async (
+    backendIdentifier: BackendIdentifier | AppId,
+    secretName: string
+  ): Promise<void> => {
+    try {
+      return await this.secretClient.removeSecret(
+        backendIdentifier,
+        secretName
+      );
+    } catch (e) {
+      throw this.translateToAmplifyError(e, 'Remove');
+    }
+  };
+
+  private translateToAmplifyError = (error: unknown, apiName: string) => {
+    if (error instanceof SecretError && error.cause) {
+      if (
+        [
+          'UnrecognizedClientException',
+          'AccessDeniedException',
+          'NotAuthorized',
+          'ExpiredTokenException',
+          'CredentialsProviderError',
+        ].includes(error.cause.name)
+      ) {
+        return new AmplifyUserError('SSMCredentialsError', {
+          message: `Failed to ${apiName} secrets. ${error.cause.name}: ${error.cause?.message}`,
```

Comment:
Nit: With `apiName`s passed in this would be "Failed to Get|List|Set|Remove secrets", do we want these to be lowercase?

Reply:
Updated!